pkgs/top-level/all-packages.nix

```diff
@@ -8420,6 +8420,8 @@ with pkgs;
 
   pyprof2calltree = pythonPackages.callPackage ../development/tools/profiling/pyprof2calltree { };
 
+  pre-commit = pythonPackages.callPackage ../development/tools/pre-commit { };
```

Comment:
Please don't use `pythonPackages` here, but as argument in the expression.

Reply:
Also, since pre-commit seems to support Python 3, please use that if possible.

Reply:
So I don't have to list individual requirements for pre-commit, just for the modules? Or am I misunderstanding something.

I'll use python3. Is the preferred method to do the following here:

```nix
  pre-commit = callPackage ../development/tools/pre-commit {
    pythonPackages = python3Packages;
  };
```

Reply:
Ye, correctly. The advantage is that applications can override individual packages.
No, the latter is not preferred. I'd just use `python3` or `python3Packages` as argument.